src/app/Dashboard/DraggableRef.tsx

```diff
@@ -0,0 +1,139 @@
+ *
+ * Subject to the condition set forth below, permission is hereby granted to any
+ * person obtaining a copy of this software, associated documentation and/or data
+ * (collectively the "Software"), free of charge and under any and all copyright
+ * rights in the Software, and any and all patent rights owned or freely
+ * licensable by each licensor hereunder covering either (i) the unmodified
+ * Software as contributed to or provided by such licensor, or (ii) the Larger
+ * Works (as defined below), to deal in both
+ *
+ * (a) the Software, and
+ * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
+ * one is included with the Software (each a "Larger Work" to which the Software
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import { dashboardCardConfigResizeCardIntent } from '@app/Shared/Redux/Configurations/DashboardConfigSlicer';
+import { gridSpans } from '@patternfly/react-core';
+import _ from 'lodash';
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { DashboardCardContext } from './ResizableCard';
+
+export interface DraggableRefProps {
+  dashboardIdx: number;
+  minimumSpan: gridSpans;
+}
+
+function normalizeAsGridSpans(val: number, min: number, max: number, a: number, b: number): gridSpans {
+  if (val < min) val = min;
+  else if (val > max) val = max;
+  const ans = Math.round((b - a) * ((val - min) / (max - min)) + a);
+  return _.clamp(ans, a, b) as gridSpans;
+}
+
+export const DraggableRef: React.FunctionComponent<DraggableRefProps> = (props) => {
+  const dispatch = useDispatch();
+
+  const cardRef = React.useContext(DashboardCardContext);
+  const draggableRef = React.useRef<HTMLDivElement>(null);
+  const isResizing = React.useRef<boolean>(false);
+  const setInitialVals = React.useRef<boolean>(true);
+
+  const SMALLEST_CARD_WIDTH = 126;
```

Comment:
Where does this constant come from?

Reply:
It's just sort of an eyeball magic number. I believe there is probably a better way to get the boundingClientRect.right of the minimum size of each Card (which is what I'm trying to compute in `minimumCardRight`), so likely this constant will get removed.

Reply:
Sounds good. Please just be careful to avoid hardcoding any assumptions that may have to do with screen dimensions like width in pixels. The assumption of 12 grid spans is fine since this is very well established in Bootstrap and there is the whole span-by-screensize css system for responsive layouts, but anything else should be generic or dynamic, or at the very least use relative css units.